packages/eslint-plugin/src/rules/strict-enums.ts

```diff
@@ -0,0 +1,678 @@
+    }
+
+    function isNullOrUndefinedOrAnyOrUnknownOrNever(
+      ...types: ts.Type[]
+    ): boolean {
+      return types.some(type =>
+        util.isTypeFlagSet(
+          type,
+          ts.TypeFlags.Null |
+            ts.TypeFlags.Undefined |
+            ts.TypeFlags.Any |
+            ts.TypeFlags.Unknown |
+            ts.TypeFlags.Never,
+        ),
+      );
+    }
+
+    function setHasAnyElementFromSet<T>(set1: Set<T>, set2: Set<T>): boolean {
+      for (const value of set2.values()) {
+        if (set1.has(value)) {
+          return true;
+        }
+      }
+
+      return false;
+    }
+
+    // --------------
+    // Main functions
+    // --------------
+
+    function checkCallExpression(
+      node: TSESTree.CallExpression | TSESTree.NewExpression,
+    ): void {
+      const tsNode = parserServices.esTreeNodeToTSNodeMap.get(node);
+      const signature = typeChecker.getResolvedSignature(tsNode);
+      if (signature === undefined) {
+        return;
+      }
+
+      /**
+       * The `getDeclaration` method actually returns
+       * `ts.SignatureDeclaration | undefined`, not `ts.SignatureDeclaration`.
+       */
+      const declaration = signature.getDeclaration();
+      if (declaration === undefined) {
+        return;
+      }
+
+      /**
+       * Iterate through the arguments provided to the call function and cross
+       * reference their types to the types of the "real" function parameters.
+       */
+      for (let i = 0; i < node.arguments.length; i++) {
+        const argument = node.arguments[i];
+        const argumentType = getTypeFromNode(argument);
+        let parameterType = signature.getTypeParameterAtPosition(i);
+
+        /**
+         * If this function parameter is a generic type that extends another
+         * type, we want to compare the calling argument to the constraint
+         * instead.
+         *
+         * For example:
+         *
+         * ```ts
+         * function useFruit<FruitType extends Fruit>(fruitType: FruitType) {}
+         * useFruit(0)
+         * ```
+         *
+         * Here, we want to compare `Fruit.Apple` to `Fruit`, not `FruitType`,
+         * because `FruitType` would just be equal to 0 in this case (and
+         * would be unsafe).
+         */
+        const parameter = declaration.parameters[i];
+        if (parameter !== undefined) {
+          const parameterTSNode = getTypeFromTSNode(parameter);
+          const constraint = parameterTSNode.getConstraint();
+          if (constraint !== undefined) {
+            parameterType = constraint;
+          }
+        }
+
+        /**
+         * Disallow mismatched function calls, like the following:
+         *
+         * ```ts
+         * function useFruit(fruit: Fruit) {}
+         * useFruit(0);
+         * ```
+         */
+        if (isMismatchedEnumFunctionArgument(argumentType, parameterType)) {
+          context.report({
+            node,
```

Comment:
could we report on the argument instead of the entire call expression here?
this would allow us to be as specific as possible with the highlighting and make the rule look less noisy if the call expression is multi-lined.

```suggestion
            node: node.arguments[i],
```

Reply:
i commited the suggestion and the tests still pass, so i assume yes.